src/coreclr/vm/excep.cpp

```diff
@@ -3374,25 +3376,30 @@ BOOL StackTraceInfo::AppendElement(BOOL bAllowAllocMem, UINT_PTR currentIP, UINT
     }
 
 #ifndef TARGET_UNIX // Watson is supported on Windows only
-    Thread *pThread = GetThread();
-
-    if (pThread && (currentIP != 0))
+#ifdef FEATURE_EH_FUNCLETS
+    if (!g_isNewExceptionHandlingEnabled)
+#endif // FEATURE_EH_FUNCLETS
```

Comment:
Could you describe the impact of your change on Watson bucketing? It appears that the ExceptionTracker was involved in the bucketing system, but the ExInfo logic that you've integrated is not. In addition, my reading of this is that we may be calling the "FallbackFailFast" function when there is an unhandled exception, and I don't know how that interacts with the Watson behavior. I'm particularly concerned about triage dumps and ensuring that they continue to contain the correct details about what exception was unhandled, and where it was thrown from.

Reply:
The watson stuff is not completed yet in this first version. I've mentioned that debugging stuff is not fully working in the PR description and the watson is similar case. I will work on these in the next step either for some .NET 8 update or .NET 9. These are the main reasons why it is not enabled by default.